agenta-backend/agenta_backend/services/evaluators_service.py

```diff
@@ -393,33 +396,59 @@ async def auto_ai_critique(
 
 async def ai_critique(input: EvaluatorInputInterface) -> EvaluatorOutputInterface:
     openai_api_key = input.credentials.get("OPENAI_API_KEY", None)
-
+    anthropic_api_key = input.credentials.get("ANTHROPIC_API_KEY", None)
+    litellm.openai_key = openai_api_key
+    litellm.anthropic_api_key = anthropic_api_key
     if not openai_api_key:
         raise Exception(
             "No OpenAI key was found. AI Critique evaluator requires a valid OpenAI API key to function. Please configure your OpenAI API and try again."
         )
-
-    chain_run_args = {
-        "llm_app_prompt_template": input.inputs.get("prompt_user", ""),
-        "variant_output": input.inputs["prediction"],
-        "correct_answer": input.inputs["ground_truth"],
-    }
-    for key, value in input.inputs.items():
-        chain_run_args[key] = value
-
-    prompt_template = input.settings.get("prompt_template", "")
-    messages = [
-        {"role": "system", "content": prompt_template},
-        {"role": "user", "content": str(chain_run_args)},
-    ]
-
-    print(input)
-
-    client = AsyncOpenAI(api_key=openai_api_key)
-    response = await client.chat.completions.create(
-        model="gpt-3.5-turbo", messages=messages, temperature=0.8
-    )
-    evaluation_output = response.choices[0].message.content.strip()
+    if (
+        input.settings.get("version", "1") == "2"
```

Comment:
What's our plan for this ?
To support the old version forever ?
No need to answer now. But it points to a more general question or rolling support for old features and entities.